librz/util/buf.c

```diff
@@ -1368,9 +1412,22 @@ RZ_API ut64 rz_buf_fwd_scan(RZ_NONNULL RzBuffer *b, ut64 start, ut64 amount, RZ_
 	if (!amount) {
 		return 0;
 	}
-	if (b->methods->get_whole_buf) {
-		ut64 sz;
-		const ut8 *buf = b->methods->get_whole_buf(b, &sz);
+	ut64 sz;
+	ut8 *buf;
+	if (b->type == RZ_BUFFER_BYTES || b->type == RZ_BUFFER_MMAP) {
+		if (b->type == RZ_BUFFER_BYTES) {
+			buf = buf_bytes_get_whole_buf(b, &sz);
+		} else {
+			buf = buf_mmap_get_whole_buf(b, &sz);
+		}
+		if (buf) {
```

Comment:
You can invert the condition and return early with an error.

Reply:
Also below please.

Reply:
LGTM once this is resolved.